ricecooker/managers/tree.py

```diff
@@ -33,71 +33,33 @@ def process_tree(self, node, parent=None):
                 parent (Node): parent of node being processed
             Returns: None
         """
-        from ..classes import nodes
-
-        # If node is not a channel, download files
-        if isinstance(node, nodes.Channel):
-            if node.thumbnail is not None and node.thumbnail != "":
-                file_data = config.DOWNLOADER.download_file(node.thumbnail, "Channel Thumbnail", default_ext=file_formats.PNG)
-                node.thumbnail = file_data['filename'] if file_data else ""
-        else:
-            node.files = config.DOWNLOADER.download_files(node.files, "Node {}".format(node.source_id))
-
-            # Get the thumbnail if provided or needs to be derived
-            thumbnail = None
-            if node.thumbnail is not None:
-                thumbnail = config.DOWNLOADER.download_file(node.thumbnail, "Node {}".format(node.source_id), default_ext=file_formats.PNG, preset=node.thumbnail_preset)
-            elif isinstance(node, nodes.Video) and node.derive_thumbnail:
-                thumbnail = config.DOWNLOADER.derive_thumbnail(config.get_storage_path(node.files[0]['filename']), "Node {}".format(node.source_id))
-
-            if thumbnail:
-                node.files.append(thumbnail)
-
-            # If node is an exercise, process images for exercise
-            if isinstance(node, nodes.Exercise):
-                node.process_questions()
+        downloaded = node.process_files()
 
         # Process node's children
         for child_node in node.children:
-            self.process_tree(child_node, node)
+            downloaded += self.process_tree(child_node, node)
+
+        return list(filter(lambda x: x, set(downloaded))) # Remove any duplicate or null files
```

Comment:
Filters can be nice, but for some things, list comprehensions are more readable. This could also be:
`return [x for x in set(downloaded) if x]`